docs/prober-and-monitoring.md

```diff
@@ -0,0 +1,81 @@
+# Prober and monitoring
+
+**PMAP is not a official Google product.**
+
+Monitoring consists of two parts: prober and alerting.
+
+## Prober
+
+[Prober](../prober/) is a go binary being deployed to [cloud run
+job](https://cloud.google.com/run/docs/overview/what-is-cloud-run#jobs), and is
+triggered by [cloud scheduler](https://cloud.google.com/scheduler) to constantly
+probing pmap services to check if the services are up.
+
+In each prober job, the prober will upload a yaml file to
+[gcs](https://cloud.google.com/storage), and query the bigquery table to see if
+a corrosponding entry exists in bigquery table. The job is considered as success
+on if the following requirements for both mapping and policy service are met:
+
+1. The object is uploaded.
+2. There is an entry in bigquery table that matches the traceID.
+
+## Monitoring and Alert Policies
+
+We monitor the pmap service with [native cloud run monitoring
+metrics](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-run) and set up
+the alert for all pubsub subscriptions base on the following metrics:
+
+- subscription/oldest_unacked_message_age
+- subscription/dead_letter_message_counts
+
+We also have alert base on the prober job execution result. The metrics used is:
+
+- job/completed_execution_count
+
+## Installation
+
+You can use the provided Terraform module to setup Prober, or you can refer to
+the provided module to build it from scratch.
+
+```terraform
+module "prober_and_monitoring" {
+  source = "../modules/monitoring"
+
+  project_id = var.project_id
+
+  prober_bucket_id           = "GCS bucket id for where objects are uploaded to."
```

Comment:
Make it clear which variables are users that need to provide. Currently, it doesn't look clear.

Reply:
Updated